### PR TITLE
SDK-599 -- Add ability to launch Windows application from web link

### PR DESCRIPTION
**Pull request by apeterson-branch**
## Reference
SDK-599 -- Add ability to launch Windows application from web link

## Description
We need to be able to click on a web link, and have a Windows sample application open.

After this change, the ColorPicker sample responds to a URI that looks like color://foo/bar

*HOWEVER*
This works by creating registry keys under HKEY_CURRENT_USER\Software\Classes\color.  To set up the registry keys, I added a "File -> Install" menu option that does the right thing.

## Testing Instructions
1. Build and run the latest Windows ColorPicker Sample
2. Dismiss the color dialog that opened the first time
3. Select File -> Install
4. Exit the app.
5. (optional) Examine the Registry to see the pieces that were required to make this work
6. In Chrome, type in a url that looks like color://foo/bar

Expected:  The Color Picker will open (after it prompts you the first time if that is what you want)

## Risk Assessment [`MEDIUM`]
This is fairly self-contained.  It will either work or not. 

- [x] I, the PR creator, have tested — integration, unit, or otherwise — this code.

## Reviewer Checklist (To be checked off by the reviewer only)

- [ ] JIRA Ticket is referenced in PR title.
- Correctness & Style
    - [ ] Conforms to [Style Guides](https://google.github.io/styleguide/cppguide.html)
    - [ ] Mission critical pieces are documented in code and out of code as needed.
- [ ] Unit Tests reviewed and test issue sufficiently.
- [ ] Functionality was reviewed in QA independently by another engineer on the team.

### CODE REVIEW

File: BranchSDK-Samples/Windows/ColorPicker/ColorPicker.cpp [hunk 1]
```diff
@@ -226,6 +230,71 @@ INT_PTR CALLBACK About(HWND hDlg, UINT message, WPARAM wParam, LPARAM lParam)
...
+void installApp()
+{
+    int bytesize = sizeof(TCHAR);
+
+    HKEY hKey;
+    LPCTSTR keyBase = TEXT("SOFTWARE\\Classes\\color");
+
+    // Step 1:  Create the Base Key and default values
+    LONG rc = RegCreateKeyEx(HKEY_CURRENT_USER, keyBase, 0, NULL, REG_OPTION_NON_VOLATILE, KEY_WRITE, NULL, &hKey, NULL);
+    if (rc != ERROR_SUCCESS) {
```
Comment (bklastaitis-branch) [inline BranchSDK-Samples/Windows/ColorPicker/ColorPicker.cpp]: So which is it, error or success? 😄(naming of ERROR_SUCCESS)

File: BranchSDK-Samples/Windows/ColorPicker/ColorPicker.cpp [hunk 2]
```diff
@@ -226,6 +230,71 @@ INT_PTR CALLBACK About(HWND hDlg, UINT message, WPARAM wParam, LPARAM lParam)
...
+
+    LPCTSTR baseDefaultValue = TEXT("URL:Color Picker");
+    rc = RegSetValueEx(hKey, TEXT(""), 0, REG_SZ, (LPBYTE)baseDefaultValue, BYTESIZE(_tcslen(baseDefaultValue) + 1));
+
+    LPCTSTR keyProtocol = TEXT("URL Protocol");
+    LPCTSTR valProtocol = TEXT("");
+    rc = RegSetValueEx(hKey, keyProtocol, 0, REG_SZ, (LPBYTE)valProtocol, BYTESIZE(_tcslen(valProtocol) + 1));
+
+    // Step 2:  Set the executable location
+    HMODULE hModule = GetModuleHandleW(NULL);
```
Comment (bklastaitis-branch) [inline BranchSDK-Samples/Windows/ColorPicker/ColorPicker.cpp]: Should this be inside the ifdef block?

Comment (apeterson-branch) [reply BranchSDK-Samples/Windows/ColorPicker/ColorPicker.cpp]: Good catch!  Fixed.

File: BranchSDK-Samples/Windows/ColorPicker/ColorPicker.cpp (hunk 1 above)
Comment (apeterson-branch) [reply BranchSDK-Samples/Windows/ColorPicker/ColorPicker.cpp]: That's a Windows definition.
Reference:  https://docs.microsoft.com/en-us/windows/win32/debug/system-error-codes--0-499-

**Pull request merged**